server/app/services/applicant/ApplicantService.java

```diff
@@ -608,18 +614,21 @@ private ApplicationPrograms relevantProgramsForApplicant(
         Sets.difference(activeProgramNames.keySet(), programNamesWithApplications);
     unappliedActivePrograms.forEach(
         programName -> {
+          OptionalInt autoFillCount = getAutoFillCount(applicantId, activeProgramNames.get(programName).id()).toCompletableFuture().join();
```

Comment:
suggestion (blocking): this is moving in the right direction, but the `join` calls here and  on L579 block the HTTP thread.